app/models/filters/criteria/filter_for_projects.rb

```diff
@@ -0,0 +1,27 @@
+class Filters::Criteria::FilterForProjects < Filters::Criteria::Base
+  def applies?
+    input.project_ids.present? || input.project_group_ids.present?
+  end
+
+  def apply(scope)
+    project_ids = visible_project_ids
+    return scope if project_ids.blank?
+
+    # seems to be order dependent (merge before in_project)
+    scope.merge(viewable_project_scope).in_project(project_ids)
+  end
+
+  protected
+
+  def visible_project_ids
+    project_ids = []
+    project_ids += input.project_ids || [] if user.report_filter_visible?(:project_ids)
+
+    # note, this is from the original logic: filter by groups even if user cannot filter by project_id
```

Comment:
```suggestion
    # Note: since filtering by project groups amounts to filtering by sets of project ids, and they need to be OR'd with projects so that they aren't limiting to overlaps with chosen projects, we extract the project_ids from the chosen project groups and add them to those from the projects input.
```

This is interesting, and not something I remembered.  I see why it works this way, but I can also see a solid argument for breaking project groups out into their own criterion.  
Looking at `FilterBase#effective_project_ids` the inputs are all OR'd, and very few reports use `anded_effective_project_ids`.  
Given that, I guess I'd be inclined to move ProjectGroups out of this filter for projects for consistency.  What do you think?

Reply:
Thanks for the note, I'm not quite clear what effective_project_ids or anded_effective_project_ids are doing in filter base. Let's try and discuss when we huddle next

In principal I agree that breaking these into additional criteria makes sense. However I think I need some additional context. I was hoping to limit scope of this refactor (it's already pretty wide already) so I'm a little cautious about splitting the filter

Reply:
added some comments to filter base to document effective_project_ids; it's used for project-scoped reports, not enrollments.

I believe we decided to leave the filter as-is to avoid changing behavior